### PR TITLE
[Compute] `az restore-point`: Add new parameters to support cross region copy

### PR DESCRIPTION
**Pull request by PARADISSEEKR**
**Description**

Changes needed in az-restore-point and az-restore-point collection cmdlets to support cross region copy https://github.com/Azure/azure-cli/issues/21878

**Testing Guide**

**History Notes**

[Compute] `az restore-point create`: Add a new parameter `--source-restore-point` to support cross region copy
[Compute] `az restore-point show`: Add a new parameter `--instance-view` to show the instance view of a restore point and replace the deprecated `--expand`
[Compute] `az restore-point collection show`: Add a new parameter `--restore-points` to show all contained restore points in the restore point collection and replace the deprecated `--expand`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -1464,4 +1468,9 @@ def load_arguments(self, _):
     with self.argument_context('restore-point collection update') as c:
         c.argument('tags', tags_type)
 
+    with self.argument_context('restore-point collection show') as c:
+        c.argument('expand', help='The expand expression to apply on the operation.',
+                   deprecate_info=c.deprecate(hide=True))
+        c.argument('restore_points', action='store_true', help='All contained restore points in the restorePointCollection.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:1474]:
```suggestion
        c.argument('restore_points', action='store_true', help='Show all contained restore points in the restorePointCollection.')
```

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -1439,10 +1439,14 @@ def load_arguments(self, _):
...
                    'customer wishes to exclude from the restore point. If no disks are specified, all disks will be '
                    'included.')
+        c.argument('source_restore_point', help='Resource Id of the source restore point from which a copy needs to be created')
 
     with self.argument_context('restore-point show') as c:
         c.argument('restore_point_name', options_list=['--name', '-n', '--restore-point-name'],
                    help='The name of the restore point.')
+        c.argument('expand', help='The expand expression to apply on the operation.',
+                   deprecate_info=c.deprecate(hide=True))
+        c.argument('instance_view', action='store_true', help='The instance view of a restore point.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:1449]:
```suggestion
        c.argument('instance_view', action='store_true', help='Show the instance view of a restore point.')
```

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 3]
```diff
@@ -1464,4 +1468,9 @@ def load_arguments(self, _):
     with self.argument_context('restore-point collection update') as c:
         c.argument('tags', tags_type)
 
+    with self.argument_context('restore-point collection show') as c:
+        c.argument('expand', help='The expand expression to apply on the operation.',
+                   deprecate_info=c.deprecate(hide=True))
+        c.argument('restore_points', action='store_true', help='Show all contained restore points in the restorePointCollection.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:1474]:
```suggestion
        c.argument('restore_points', action='store_true', help='Show all contained restore points in the restore point collection.')
```

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**